src/sorted_vec.h

```diff
@@ -0,0 +1,93 @@
+#ifndef SORTED_SET_H
+#define SORTED_SET_H
+
+#include <algorithm>
+#include <cassert>
+#include <vector>
+
+/// Modifying a key in associative containers leads to UB. This is an std::set
+/// reimplementation with worse characteristics (such as re-sorting the
+/// container on every push), but one that allows for elements to be
+/// out-of-order (which can happen after a key was modified)
+template<class Elem, class Compare = std::less<Elem> >
+class SortedVec {
+  vector<Elem> data;
```

Comment:
Please move private at the end of the class. https://google.github.io/styleguide/cppguide.html#Declaration_Order